src/agentscope/service/service_toolkit.py

```diff
@@ -195,6 +195,10 @@ def bing_search(query: str, api_key: str, num_results=10):
 
         """
 
+        if isinstance(service_func, partial):
```

Comment:
Why do we need to consider the partial function here?

Reply:
For example, api_key should be passed in bing_serach.
I am not sure if there is another way to implement the following in `workflow_node.py`
 `service_toolkit.add(bing_search, api_key=BING_API_KEY, num_results=3)`

Reply:
add TODO here